common/src/main/java/com/wynntils/wc/utils/scoreboard/ScoreboardManager.java

```diff
@@ -67,30 +68,37 @@ public final class ScoreboardManager {
         }
 
         List<ScoreboardLine> scoreboardCopy = new ArrayList<>(reconstructedScoreboard);
+        LinkedList<ScoreboardLineChange> queueCopy = new LinkedList<>(queuedChanges);
 
-        for (ScoreboardLineChange queuedChange : queuedChanges) {
-            if (queuedChange.method() == ServerScoreboard.Method.REMOVE) {
-                scoreboardCopy.removeIf(
-                        scoreboardLine -> Objects.equals(scoreboardLine.line(), queuedChange.lineText()));
-            }
+        Map<Integer, ScoreboardLine> scoreboardLineMap = new TreeMap<>();
+
+        for (ScoreboardLine scoreboardLine : scoreboardCopy) {
+            scoreboardLineMap.put(scoreboardLine.index(), scoreboardLine);
         }
 
         Set<String> changedLines = new HashSet<>();
-        while (!queuedChanges.isEmpty()) {
-            ScoreboardLineChange processed = queuedChanges.pop();
+        while (!queueCopy.isEmpty()) {
+            ScoreboardLineChange processed = queueCopy.pop();
 
             if (processed.method() == ServerScoreboard.Method.REMOVE) {
+                for (ScoreboardLine lineToRemove : scoreboardLineMap.values().stream()
+                        .filter(scoreboardLine -> Objects.equals(scoreboardLine.line(), processed.lineText()))
+                        .toList()) {
+                    scoreboardLineMap.remove(lineToRemove.index());
+                }
                 continue;
             }
```

Comment:
how about removing continue and making this into a proper else part? I think it would make it clearer -- if we're removing, do one thing, otherwise add/update the content.